s3repo/model.py

```diff
@@ -287,6 +346,49 @@ def _get_repository_list(self):
 
         return repos_list
 
+    @staticmethod
+    def _get_abs_path(base_path, path):
+        """Get absolute (base_path + path) normalized path.
+        """
+
+        ret = path
+        if base_path:
+            ret = os.path.normpath('/'.join([base_path, ret]))
+        return ret
+
+    def determine_type_and_path(self, path):
+        """Find an object spcified by "path" and determine its
+        type and absolute path. Type can be "directory", "file" or
+        None if object hasn't found."""
+
+        base_path = self.s3_settings.get('base_path') or ''
+        abs_path = S3AsyncModel._get_abs_path(self.s3_settings.get('base_path'), path)
+        abs_path = abs_path.strip('/')
+
+        # Parameters for list_objects_v2():
+        # * 'Bucket' is a bucket name.
```

Comment:
We usually used double quotes earlier in the description. Please replace it. Here and below.

Reply:
Done.